amd/src/data.js

```diff
@@ -0,0 +1,263 @@
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Container for tiny_elements data (categories, components, flavors, variants).
+ *
+ * @module     tiny_elements/data
+ * @copyright  2025 ISB Bayern
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+import {get_strings as getStrings} from 'core/str';
+import {component as pluginname} from './common';
+import {
+    variantExists,
+    setData as setVariantsData
+} from './variantslib';
+import {
+    findById,
+    findByName
+} from './helper';
+import {call as fetchMany} from 'core/ajax';
+
+export default class Data {
+    categories = [];
+    components = [];
+    flavors = [];
+    variants = [];
+    langStrings = {};
+    userStudent = false;
+    canManage = false;
+    contextid = 1;
```

Comment:
let uninitialized or 0 or something that is obviously wrong and will be overwritten :)

Reply:
This would look weird to me - I'd like to keep it like that.